File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {// [1,8,6,2,5,4,8,3,7]
        int maxi = 0, left = 0, right = height.size()-1;// left = 1 right = 7
        while(left < right){
            maxi = max(maxi, (right - left) * min(height[left], height[right]));// maxi = 49
            if(height[left] <  height[right]) left++;
            else right--;
        }
        return maxi;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    private int min(int a, int b){
        if(a<b)return a;
        else return b;
    }
    private int max(int a, int b){
        if(a<b)return b;
        else return a;
    }
    public int maxArea(int[] height) {
        int n = height.length;
        int l=0,r=n-1;
        int ans=0;
        if(n==2){
            ans=min(height[0],height[1]) * min(height[0],height[1]);
            return ans;
        }
        while(l<r){
            ans=max(ans, min(height[l],height[r]) * (r-l));
            if(height[l]<height[r])l++;
            else r--;
        }
        return ans;
    }
}